### PR TITLE
add image building to pr gha workflow

### PR DESCRIPTION
**Pull request by gflarity**
This PR builds includes building the docker image(s) as part of the the PR GitHub Action workflow. I fixed a bug in the Dockerfile recently that caused the nightly to fail, but not the PR tests. This change will catch such an issue at the PR stage.

### CODE REVIEW

File: .github/workflows/pr_build.yaml [hunk 1]
```diff
@@ -162,3 +162,25 @@ jobs:
...
+        with:
+          go-version-file: 'go.mod'
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Install regctl
+        uses: regclient/actions/regctl-installer@main
+      - name: Build image
+        run: make docker-build
```
Comment (faisal-memon) [inline .github/workflows/pr_build.yaml:186]: Can we add a step here to save the images that are built, similar to this? https://github.com/spiffe/spire/blob/main/.github/workflows/pr_build.yaml#L186-L192

Comment (kfox1111) [reply .github/workflows/pr_build.yaml:186]: That basically builds a tar of the same images uploaded to the registry. What is intent of the extra copy?

Maybe it should be removed from the spiffe/spire repo too.

A user can make the tarball for offline access themselves with docker, podman, skopeo, or crane pretty easily.

Comment (faisal-memon) [reply .github/workflows/pr_build.yaml:186]: valid point

File: .github/workflows/pr_build.yaml [hunk 2]
```diff
@@ -162,3 +162,25 @@ jobs:
...
+      - name: Setup go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Install regctl
+        uses: regclient/actions/regctl-installer@main
```
Comment (faisal-memon) [inline .github/workflows/pr_build.yaml:184]: I dont think we need regctl just to build the image.

Comment (faisal-memon) [reply .github/workflows/pr_build.yaml:184]:
```suggestion
```

**Review by faisal-memon — APPROVED** [review]
Thanks @gflarity

**Pull request merged**